File: system-test/java/src/main/java/com/optivem/shop/testkit/driver/port/external/erp/ErpDriver.java
package com.optivem.shop.testkit.driver.port.external.erp;

import com.optivem.shop.testkit.driver.port.external.erp.dtos.GetProductRequest;
import com.optivem.shop.testkit.driver.port.external.erp.dtos.GetProductResponse;
import com.optivem.shop.testkit.driver.port.external.erp.dtos.ReturnsProductRequest;
import com.optivem.shop.testkit.driver.port.external.erp.dtos.ReturnsPromotionRequest;
import com.optivem.shop.testkit.driver.port.external.erp.dtos.error.ErpErrorResponse;
import com.optivem.shop.testkit.common.Result;

public interface ErpDriver extends AutoCloseable {
    Result<Void, ErpErrorResponse> goToErp();

    Result<GetProductResponse, ErpErrorResponse> getProduct(GetProductRequest request);

    Result<Void, ErpErrorResponse> returnsProduct(ReturnsProductRequest request);

    Result<Void, ErpErrorResponse> returnsPromotion(ReturnsPromotionRequest request);
}

File: system-test/typescript/src/testkit/driver/port/external/erp/erp-driver.ts
import { Result } from '../../../../common/result.js';
import { ErpErrorResponse } from './dtos/ErpErrorResponse.js';
import { GetProductRequest } from './dtos/GetProductRequest.js';
import { GetProductResponse } from './dtos/GetProductResponse.js';
import { ReturnsProductRequest } from './dtos/ReturnsProductRequest.js';
import { ReturnsPromotionRequest } from './dtos/ReturnsPromotionRequest.js';

export interface ErpDriver {
  goToErp(): Promise<Result<void, ErpErrorResponse>>;
  getProduct(request: GetProductRequest): Promise<Result<GetProductResponse, ErpErrorResponse>>;
  returnsProduct(request: ReturnsProductRequest): Promise<Result<void, ErpErrorResponse>>;
  returnsPromotion(request: ReturnsPromotionRequest): Promise<Result<void, ErpErrorResponse>>;
  close(): Promise<void>;
}
